src/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializer.Write.HandleEnumerable.cs

```diff
@@ -42,16 +53,9 @@ public static partial class JsonSerializer
                 {
                     writer.WriteStartArray(jsonPropertyInfo._name);
                 }
-
-                IEnumerable enumerable = (IEnumerable)jsonPropertyInfo.GetValueAsObject(state.Current.CurrentValue, options);
-
-                if (enumerable != null)
-                {
-                    state.Current.Enumerator = enumerable.GetEnumerator();
-                }
             }
 
-            if (state.Current.Enumerator != null && state.Current.Enumerator.MoveNext())
```

Comment:
Is it guaranteed not to be null? Add a debug.assert?

Reply:
Yes because the code above it now sets `state.Current.Enumerator = enumerable.GetEnumerator()`

Reply:
And `enumerable.GetEnumerator()` is guaranteed to not return null?